tests/ui/verbose_file_reads.rs

```diff
@@ -0,0 +1,30 @@
+#![warn(clippy::verbose_file_reads)]
+use std::env::temp_dir;
+use std::fs::File;
+use std::io::Read;
+
+struct Struct;
+// To make sure we only warn on File::{read_to_end, read_to_string} calls
+impl Struct {
+    pub fn read_to_end(&self) {}
+
+    pub fn read_to_string(&self) {}
+}
+
+fn main() -> std::io::Result<()> {
+    let mut path = temp_dir();
+    path.push("test.txt");
+    let file = File::create(&path)?;
```

Comment:
This shouldn't be necessary to test the lint, since ui-test doesn't get executed by default.